12-watchtower-API.md

```diff
@@ -4,245 +4,337 @@
+2. data:  
+	* [`u16 `:`receipt_len`]
+	* [`receipt_len*byte `: `receipt`]
+	* [`u16`: `wt_signature_algorithm`]
+	* [`u16`: `wt_signature_len`
+	* [`wt_signature_len*byte`: `wt_signature`]
+	* [`u16`: `wt_public_key_len`]
+	* [`wt_public_key_len*byte`: `wt_public_key`]
 
-`ceil(maximum_viable_transaction_size / cipher_block_size`) 
+The server:
 
-`minimum_viable_transaction_size` and `maximum_viable_transaction_size` refer to the minimum/maximum size required to create a valid transaction.
+* MUST set `receipt` to a receipt built according to 	[Receipt-Format](#receipt-format).
+* MUST set `wt_signature_algorithm` to one of the signature algorithms he has announced.
+* MUST set `wt_signature` to the signature of the appointment using `wt_signature_algorithm`.
+* MUST set `wt_public_key` to the public key that matches the private key used to create `wt_signature`.
 
-Accepting `encrypted_blob` outside those boundaries will ease DoS attackes on the server.
- 
+The client:
 
-## Range of values 
+* MUST compute the `wt_signature` verification using `wt_public_key`.
 
-- ```start_block```: (Absolute) Block number 
-- ```end_block```: (Absolute) Block number
-- ```dispute_delta```: (Relative) Block number
-- ```transaction_size```: Measured in Bytes (e.g. 200) 
-- ```transaction_fee```: Measured in sats (e.g. 2000)
-- ```cipher```: AESGCM256, CHACHA20
-- ```customer_signature_algorithm```: ECDSA, SCHNORR
+* MUST fail the connection if:
+	* Any of the fields is missing.
+	* `receipt` does not matches the format specified at 	[Receipt-Format](#receipt-format)
+	* `receipt` fields do not match the ones sent in the `appointment` message.
+	* `wt_signature_algorithm` does not match any of the ones offered by the server.
+	* `wt_signature` cannot be verified using `wt_public_key`.
 
-### Rationale
+#### Receipt Format 
 
-We can group the data fields into logical groups. 
+The server MUST create the receipt containing the following information:
 
-* **Appointment information**: The appointment time is defined using the ```start_block``` and```end_block```. WatchTower will delete the job when the appointment has expired. We recommend only using block numbers as that is the natural clock for Bitcoin. 
-* **Explicit acknowledgement of transaction details**: The ```dispute_delta```, ```transaction_size```, and ```transaction_fee``` let the WatchTower confirm the transaction is "reasonable" and it can be accepted into the blockchain (especially if there is congestion in the future). 
-* **Encrypted transaction**: The ```cipher```,```encrypted_blob``` states how the WatchTower can later find the dispute transaction and decrypt the justice transaction. 
-* **Customer signature** The ```customer_public_key``` and ```customer_signature``` provides an explicit message about the job from the customer. 
+	txlocator
+	start_block
+	end_block
+	dispute_delta
+	encrypted_blob
+	transaction_size
+	transaction_fee
+	cipher
+	customer_public_key
```

Comment:
i thought we got rid of the customer public key.

Reply:
I don't think it's absolutely necessary. The signature should do.